### ISSUE TITLE
Progress report 3.0.1

### ISSUE THREAD

**Issue by bsfa** [SevenBits/Mac-Linux-USB-Loader#58]
The wizard does show up when clicking the first button. 
The USB flash drive does not show up in Choose destination. Should it be anything else than Fat32? 

Apparently not. After starting up MLU again, MLU does recognize the flash drive. 

On the next page, only the bottom “older” tick was set, not the top one for Mac. I figure I have to set the tick myself.

I was confused by the Grant access sheet that showed me lots of files&folders, but apparently all that was required is clicking the blue Grant access button. 

Subsequently I was greeted with a sheet that said I didn’t have permission to save the file boot in the folder efi. I run my account in non-admin mode for security reasons.  There was no option to grant the access.

I gave myself admin permissions and restarted the iMac for that to take effect. 

Still, I got the same message that I didn’t have permission to save the file.

I formatted the flash drive anew. Because the previous try MLU didn’t ask me for the .iso, I restarted MLU also and tried again. Now I could install. Progress!

The installation was finished and I pressed Okay.

I pressed the Next button. Nothing happens. I more or less expected to be brought back to the Welcome screen. 

Reading the General instructions on the web, I learn that I should restart (with the option key pressed). LMU could have told me that, with or without pressing the Next button.

I restart with Option key pressed, select the proper icon and then the installation fails. Error messages flash in red across the screen. To view them I record them using my phone. They read:

Welcome to Enterprise! Versiosn 0.2.1
Error can’t find configuration file.
Cannot continue because core files are missing or damaged
Restarting….

I don’t know whether this has to do with the USB hard drive (also Fat 32 formatted) not showing up in MLU (it can be seen in the Finder, though). 

When I try the USB set up button, select my USB flash drive, the Enable button becomes active but when I click it it says that it can’t be used for start-up. Should it be because of the .iso? So, is the problem caused by the format of the USB drive and if so, MLU could test that before putting the .iso on it.

Bert

**Comment by bsfa** [comment]
On 31 Dec, 2014, at 20:32, SevenBits <notifications@github.com> wrote:

Thank you for your reply.


> @bsfa What files are on your USB drive?

I get easily confused between the flash drive and the hard disk (both are USB). I think you’re referring to the USB flash drive here, not the hard disk.

> If I go into Terminal with a USB drive that I created, here's what I do:
> 
> : cd /Volumes/myusbname/efi/boot/
> : ls -a
> .                ..               .MLUL-Live-USB   ._.MLUL-Live-USB ._boot.efi       ._boot.iso       ._bootX64.efi    boot.efi         boot.iso         bootX64.efi
> : cat .MLUL-Live-USB
> #This file is machine generated. Do not modify it unless you know what you are doing.
> 
> entry Ubuntu
> family Ubuntu
> :

I get this:
Name-Removed-iMac-4:UNTITLED bert$ cd /Volumes/UNTITLED/efi/boot/
Name-Removed-iMac-4:boot bert$ ls -a
.		._boot.efi	._bootx64.efi	boot.iso
..		._boot.iso	boot.efi	bootx64.efi
Name-Removed-iMac-4:boot bert$ cat .MLUL-Live-USB
cat: .MLUL-Live-USB: No such file or directory
Name-Removed-iMac-4:boot bert$ 


> You should have a layout similar to this. If not, I suspect that you did not give Mac Linux USB Loader permissions to view your drive properly.

With MLUL, I expect it to take care of everything for me, or warn me if something is not OK, being part of the app’s target audience of people who are exploring unknown territory. 

I don’t know the extent of your ambitions with MLUL. If they include making it easy for regular Mac users to explore Linux, then I believe that if my account should have been an admin account, MLUL could have checked whether that was the case before starting to copy something. Or whatever needs to be done to give MLUL the permissions it needs. (I didn’t know that it needed something, and if admin is not it, then I don’t know how to give MLUL the permissions it needs). 

In a recent session, I did give my account admin privileges. I’ll reformat the USB flash drive and the USB drive to see whether that makes any difference.

Last login: Thu Jan  1 13:27:40 on ttys000
Name-Removed-iMac-4:~ ik$ cd /Volumes/UNTITLED/efi/boot/
Name-Removed-iMac-4:boot ik$ ls -a
.		._boot.efi	._bootx64.efi	boot.iso
..		._boot.iso	boot.efi	bootx64.efi
Name-Removed-iMac-4:boot ik$ cat .MLUL-Live-USB
cat: .MLUL-Live-USB: No such file or directory
Name-Removed-iMac-4:boot ik$ 

So, giving my account admin rights didn’t do the trick.

Bert

**Comment by kalihack2014** [comment]
@SevenBits How do I get Mac Linux USB Loader to recognize my USB device?

**Comment by kalihack2014** [comment]
you said something about giving permissions to the USB Loader, if I need to do that how exactly would I do that? 

Thanks!

**Comment by SevenBits** [comment]
@kalihack2014 How large is your FAT partition? There appear to be issues detecting FAT partitions that are 4 GB in size or above.

In terms giving your USB drive permissions, the first time you try to do something with a USB drive, a file dialog box pops up with a Grant Access button; you click this button while the file dialog is situated in the root folder of the USB drive to grant access. I am investigating ways to make this process simpler, FYI.

**Comment by kalihack2014** [comment]
Okay my drive is 16gb, maybe I should partition it to 4gb. I'm not exactly sure what you mean about the dialogue box with the grant access button? I'm using Mac OSX 10.9.5 Mavericks.

**Comment by kalihack2014** [comment]
@SevenBits I also partitioned my 16gb into 4 4gb partitions in MS-Dos (FAT), ExFat, Mac OSX Journaled and it doesn't recognize any of them. Am I doing something wrong?

**Comment by SevenBits** [comment]
@kalihack2014 I'm referring to this dialog:

![file](https://cloud.githubusercontent.com/assets/2482736/5593902/943e4998-91fb-11e4-8dd2-f0c0be89cec1.png)

Do you relaunch the application or refresh the list of USB drives? The list doesn't automatically update, FYI.

**Comment by kalihack2014** [comment]
Yes I do. I don't have a screen like that lol. I just format my USB and launch the Mac Linux USB Loader app.

**Comment by kalihack2014** [comment]
I even tried to do a partition that was 3gb and it still didn't recognize the USB drive. It's a 16gb but I partitioned it to 13gb and 3gb both MS-DOS (FAT) and it still doesn't recognize the drive.

**Comment by SevenBits** [comment]
@kalihack2014 It should show up when you install to a USB for the first time. If it doesn't work, you can issue this command to reset all preferences:

    defaults delete com.sevenbits.Mac-Linux-USB-Loader

I've had users who have had issues with large USB drives before. Try using a smaller USB drive, maybe 4 GB total...

**Comment by kalihack2014** [comment]
Ohh okay so this is a command you have to do after you install Kali Linux to the USB?

**Comment by SevenBits** [comment]
No, before. It will clear your preferences, should make that dialog show up on Mac Linux USB Loader 3.0 and above.

**Comment by kalihack2014** [comment]
I just issued the command, the dialouge that you showed me above is didn't show up  and still not showing any USB, I have a 4gb one I will partition and try...

**Comment by kalihack2014** [comment]
@SevenBits  Also here is an error message I got when issuing that command

Kyles-MacBook-Pro-2:~ KyleCox$ defaults delete com.sevenbits.Mac-Linux-USB-Loader
2015-01-01 20:34:26.116 defaults[2494:507] 
Domain (com.sevenbits.Mac-Linux-USB-Loader) not found.
Defaults have not been changed.
Kyles-MacBook-Pro-2:~ KyleCox$

**Comment by SevenBits** [comment]
@kalihack2014 Are you using version 3.0?

**Comment by kalihack2014** [comment]
3.0.1 OSX Macericks

**Comment by kalihack2014** [comment]
@SevenBits I also tried version 3.0 on a Yosemite Mac Mini

**Comment by kalihack2014** [comment]
@SevenBits I got the Mac Linux USB Loader to recognize a different USB device however I get an error. I will upload a picture soon.

**Comment by kalihack2014** [comment]
@SevenBits  I'm having the same problem as @bsfa

**Comment by kalihack2014** [comment]
@SevenBits Apparently the Mac Linux USB Loader doesn't create a MLUL file

Kyles-MacBook-Pro-2:efi KyleCox$ cd boot
Kyles-MacBook-Pro-2:boot KyleCox$ ls
boot.efi	boot.iso	bootx64.efi
Kyles-MacBook-Pro-2:boot KyleCox$

**Comment by SevenBits** [comment]
@kalihack2014 Use `ls -a` to see hidden files, the file is hidden by default

**Comment by kalihack2014** [comment]
@SevenBits  did that it's not there. I have my Mac setup to view all hidden files anyways ;) lol

**Comment by gorilla1977** [comment]
I have exactly the same problem with missing configuration files,
but with _4GB_ USB flash drive.
So I can confirm that large USB flash is not whats causing a problem.
Like you, I can't find those "MLUL" files...
My OS is Mavericks 10.9.5

**Comment by SevenBits** [comment]
Okay, so, is anyone having this issue (the missing config file) on Yosemite... I'm just trying to keep track of everything.

**Comment by gorilla1977** [comment]
@SevenBits sadly i'm not skilled at software debugging... if you have Yosemite installed but want to test this bug on Mavericks, you could install Parallels Desktop trial (Parallels supports Mac OS X)
And if I will be 100% confident that there's no such bug at Yosemite I will upgrade to Yosemite, despite its quite buggy (because I really need your app working)

**Comment by gorilla1977** [comment]
@SevenBits i could be wrong but this bug could be related to SBEnterpriseConfigurationWriter.m

**Comment by SevenBits** [comment]
I do have a bootable Mavericks install on a portable hard drive, but I am away and will not be able to test this until tomorrow night.

If you really need immediate assistance, you can contact me from my website; this thread is not an appropriate venue for that.

Perhaps that source file is the problem, but I am not sure. These things tend to not be that simple.

I have created dozens of live USBs on Yosemite using my tool. I have never encountered any problems.

**Comment by kalihack2014** [comment]
@gorilla1977  What distro are you creating? Ubuntu, Kali Mint, Kali linux etc.??

@SevenBits Giving this a try using Version 3.0 Mac Linux USB Loader on Yosemite creating a bootable Kali Linux USB. Will let you know how it works shortly.

**Comment by kalihack2014** [comment]
@SevenBits  Version 3.0 on Yosemite creates the "MLUL" files will test to see if it boots, like I said I'm using the Kali Linux Distro.

**Comment by kalihack2014** [comment]
@SevenBits @gorilla1977 Kali Linux will not boot from Version 3.0 using Yosemite to create it. It gets up to the boot screen on the top left and says the debian boot failed, and then open a shell, kind of like the picture I uploaded a few days ago.

![ef72c104-8ed6-11e4-8fb7-1f1769a0821f](https://cloud.githubusercontent.com/assets/9980503/5599240/0055c07e-9289-11e4-94aa-69e70570e819.JPG)

**Comment by gorilla1977** [comment]
@SevenBits @kalihack2014 I would like to make a bootable usb of Qubes OS R2 - http://qubes-os.org/
*Qubes is a security-oriented, Fedora-based desktop Linux distribution whose main concept is "security by isolation" by using domains implemented as lightweight Xen virtual machines*
http://qubes-os.org/wiki/QubesDownloads
Structure of Qubes ISO image is the same as Fedora 17 / 18
( they don't have official support for EFI-based computers, and they use ISOLINUX 4.05 for booting )
----------
I have upgraded to Yosemite and tried to create bootable USB again :
but, to my surprise, I can't see those "MLUL" files - both with 3.0.0 and 3.0.1 versions

But, when I tried older 2.0.1 version - which didn't have "Other" at "Distribution Family" list -
"MLUL" files were created perfectly but I was unable to boot because of "alloc magic is broken at"
error which happens with older Enterprise releases : https://github.com/SevenBits/Enterprise/issues/1

Of course, I tried selecting Debian/Ubuntu instead of "Other" at 3.0.0/3.0.1 versions, but it doesn't help.
----------
Summary of problem:
1) It happens in all 3.x versions, regardless of "Distribution Family" selection
2) It happens regardless if your OS X version is Mavericks or Yosemite
3) It doesn't happen in older 2.0.1 version (but there is old Enterprise problem)

I hope that this detailed information could help to track down a bug

* just for complete report : my computer is Early 2011 Macbook Pro 15" i7-2720QM
Boot ROM Version: MBP81.0047.B27 , SMC Version (system): 1.69f4 *

**Comment by kalihack2014** [comment]
@gorilla1977  so you don't get MLUL files from the 3.0 version of MAC Linux USB Loader on Yosemite? Were you set to see hidden files? You can with this command "defaults write com.apple.finder AppleShowAllFiles YES" or by doing the ls -a command in the directory. That's strange because I created a USB Kali Linux distro and got the MLUL file it just fails to boot...

**Comment by gorilla1977** [comment]
@kalihack2014 I checked hidden files with a more reliable way : `ls -alh` at Terminal ;-)
As I said I was able to see MLUL when I tried older 2.0.1 version, but can't see them with 3.0.0/3.0.1

**Comment by kalihack2014** [comment]
@SevenBits @gorilla1977 Hmmm... I just tested the USB drive I created earlier today try to boot on my Yosemite machine it boots up a screen I can't see for about half a second about video support then it just stuck saying "Welcome to Enterprise Version 0.2.1"

**Comment by gorilla1977** [comment]
@kalihack2014 are you testing the same USB drive, which gave you a failed boot from screenshot above?

**Comment by kalihack2014** [comment]
@gorilla1977 YES I am same USB.  I have no idea why it's showing MLUL files for me but not for you. It might be the distro you're using but either way I don't think anyone is ever going to find a way to boot a Kali Linux distro from USB on a Mac Machine lol
![screen shot 2015-01-02 at 6 45 34 pm](https://cloud.githubusercontent.com/assets/9980503/5601018/93356ebe-92af-11e4-9d41-2f43784a6789.png)

**Comment by kalihack2014** [comment]
@gorilla1977 What are you using the Qubes OS R2 for, the only thing I can suggest is to install it via Virtual Box, that's what I'm resorting to with Kali Linux...

**Comment by gorilla1977** [comment]
@kalihack2014 Qubes OS R2 doesn't work in Virtual Box, because of Virtual Box graphic drivers issue.
And it'd be great to have a native experience of Qubes OS, that's why I'm interested in this github project

**Comment by kalihack2014** [comment]
@gorilla1977 Gotcha! I'm going to try to get it to work in Virtual Box just for the hell of it... downloading it now. It's the first time I've heard of Qubes OS R2.

**Comment by gorilla1977** [comment]
@kalihack2014 The great value of Qubes OS is that it provides unmatched security thanks to the isolation approach. But, even if you would succeed running it in a virtual machine, then there will be no true isolation from a host operating system - Mac OS X - which could leak your information - screenshots, data dumps, etc. * And I'm not kidding: e.g. this video demonstrates one of the ways, 
how Yosemite is phoning home - http://www.washingtonpost.com/posttv/business/technology/how-apples-os-x-yosemite-tracks-you/2014/10/22/66df4386-59f1-11e4-9d6c-756a229d8b18_video.html *

**Comment by kalihack2014** [comment]
@gorilla1977  Thanks for the information! And yes I couldn't get it installed in a VM anyways. I hope Sevenbits can solve both of our problems with the next release of Mac Linux USB Loader. He said he's got some promising observations but nothing certain yet... Hopefully a Beta will be released soon!

**Comment by SevenBits** [comment]
@kalihack2014 @gorilla1977 For all of those having issues with missing configuration files... when using version 3.0.1, try manually creating the `EFI` and `BOOT` folders on your USB stick before running Mac Linux USB Loader. This comes from a suggestion from an email I received from someone who read this thread; apparently, the installation worked after that.

I'm sure that the Kali solution will be working soon... until then, try to contact me from my site directly with support requests; this thread is getting long and it is not intended to be a help forum.

**Comment by gorilla1977** [comment]
@SevenBits @bsfa @kalihack2014 After solving this issue by using the advice with folders, I got into another issue - which is slightly different, and also related to configuration file. For those who interested, please lets move there, because the situation has changed and this thread is indeed getting too long:
http://github.com/SevenBits/Mac-Linux-USB-Loader/issues/60

**Comment by kalihack2014** [comment]
@gorilla1977  what is the issue (Will move to the issue 60 forum)

**Comment by gorilla1977** [comment]
@kalihack2014 Now there's MLUL, but `Error: configuration file parsing error.` (entry & family : "Other")

**Comment by SevenBits** [comment]
Since there is now an issue dedicated to the config file problem, #63, I am closing this.

**Comment by gorilla1977** [comment]
@SevenBits @kalihack2014 @bsfa Small update about booting Qubes R2, for those who are interested:
[UEFI : Qubes R2 booting problems on Macbook Pro 8,2](http://groups.google.com/forum/#!topic/qubes-devel/NAUIW79Vfxc) In short, its impossible to boot R2, but could be possible to boot new R3 version,
when it will come out. So the case of Qubes is frozen, and won't be talked about until new version comes

**Comment by gti3993** [comment]
Well i guess before we get all complicated and geeky ,the easy fix for When you see the Grant Access PopUp Screen is just to select the usb drive not the iso or anything else ! it just needs to sudo over usb drive ! so Grant Access to usb and move on !
